### DISCUSSION TITLE
[RFC] Upstream some features from my personal ofetch fork

### DISCUSSION THREAD

**Discussion by intellild** [unjs/community#41] [Show and tell]
I hava a personal ofetch fork and I believe some feature can be upstreamed.
https://github.com/UseringOfficial/ofetch

- [middleware style hooks](https://github.com/UseringOfficial/ofetch/commit/0fe1a15eb4bc4a513ea3578d8688a01c6d6bfbb4)
- [custom parse response](https://github.com/UseringOfficial/ofetch/commit/ed17c966de7edf41cf769afd0a2929d975fb57fe?w=1#diff-eeed782a266dc1fc529c811c9b7104828cfc608f349a191cab030dba2b5bc5f5R204) to receive JSON results from LLM as stream 
- `hasBody` hook to solve some non-standard fetch polyfill, like this https://github.com/unjs/ofetch/pull/398
- [allow `$fetch.create` accept `CreateFetchOptions`](https://github.com/UseringOfficial/ofetch/commit/22dcdd6ab27080f46be63c40e47ac5fa0c729df3#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR17)
- [custom error handling](https://github.com/UseringOfficial/ofetch/commit/8b537cff59ff78bc19e36b11d209300e9b801702)
- [support custom stringify query](https://github.com/UseringOfficial/ofetch/commit/122923768ffb045912ad0875f24efdbdb584d782)

**Comment by pi0** [comment]
Seems your fork is ~25 comits behind latest version (https://github.com/UseringOfficial/ofetch/compare/main...unjs%3Aofetch%3Amain) 

if you can kindly search or create proposal issues for anything you think could be better in ofetch upstream I would love to discuss each separately. 🙏🏼
